### DISCUSSION TITLE
How can I bind a shortcut to trigger an custom function aginst the element under cursor?

### DISCUSSION THREAD

**Discussion by LintaoAmons** [ibhagwan/fzf-lua#1380] [Q&A]
I want to have a short cut to open the `buffer` element under cursor in a popup window.
So wondering how could I do it~ 
I saw there are actions can be trigger by shortcuts, but besides those predefined actions, how can I trigger a customised function with the context of selected item.

**Comment by ibhagwan** [comment]
> how can I trigger a customised function with the context of selected item.

You’d have to write a custom action with code that uses the contents of the selected item , it’s up to you how to get it, if it’s an existing buffer you can use `vim.api.nvim_buf_get_lines`, otherwise you can read the file using libuv (`vim.uv`) or `readfile()`, `:FzfLua helptags` is your friend.
